src/foam/box/SocketServiceBox.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
```

Comment:
This class shouldn't be necessary.  Take a look at the WebSocketService in foam/nanos/ws/WebSocketServer.java

It hooks up to a NanoServiceRouter which dispatches box requests to nanos services, if we do the same in SocketService then we can receive requests for all nanos services via raw sockets as well.  Then we can just host the clusteirng services as nanos services.